src/clj/rems/application/model.clj

```diff
@@ -667,6 +673,19 @@
                           (for [command (:disable-commands (get-config))]
                             {:permission command}))))
 
+(defn- get-redacted-attachment-ids [application]
+  (->> (:application/events application)
+       (filter (comp #{:application.event/attachments-redacted} :event/type))
+       (mapcat :application/redacted-attachments)
+       (map :attachment/id)
+       set))
+
+(defn- mask-redacted-attachments [attachments redacted-ids]
+  (for [attachment attachments
+        :let [redacted? (some #{(:attachment/id attachment)} redacted-ids)]]
+    (assoc-some attachment :attachment/filename (when redacted?
+                                                  "redacted"))))
```

Comment:
Is the value `"redacted"` going to show up in the UI? I guess we must translate it in that case in the UI. :thinking:

Reply:
yes, that was the original intention. localization is a good idea for the client. we also need to make sure that the original filename is visible only to the user who owns the attachment. this requires some refactoring, as a few places have already relied solely in the attachment db data instead of application model.